app/src/main/java/com/spascoding/taskycourse/feature_agenda_screen/presentation/agenda/AgendaViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package com.spascoding.taskycourse.feature_agenda_screen.presentation.agenda
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.spascoding.taskycourse.feature_register_screen.data.local.model.UserInfoManager
+import com.spascoding.taskycourse.feature_register_screen.data.local.model.userInfo
+import com.spascoding.taskycourse.feature_register_screen.domain.use_case.AuthenticationUseCases
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class AgendaViewModel @Inject constructor(
+    private val authenticationUseCases: AuthenticationUseCases,
+    private val userInfoManager: UserInfoManager,
+) : ViewModel() {
+
+    fun onEvent(event: AgendaEvent) {
+        when (event) {
+            is AgendaEvent.LogoutAction -> {
+                viewModelScope.launch {
+                    userInfoManager.userInfoFlow.collect { userInfo ->
+                        authenticationUseCases.logoutUser.invoke(userInfo.accessToken) {
```

Comment:
Here it would be a job of the use case to implement what logging out means on a high-level. It therefore doesn't really need to get the token as an argument but can rather read it from prefs itself